neard/src/runtime/mod.rs

```diff
@@ -1176,125 +1176,105 @@ impl RuntimeAdapter for NightshadeRuntime {
+                    block_height,
+                    block_hash: *block_hash,
+                })
             }
             QueryRequest::CallFunction { account_id, method_name, args } => {
                 let mut logs = vec![];
                 let (epoch_height, current_protocol_version) = {
                     let mut epoch_manager =
                         self.epoch_manager.as_ref().write().expect(POISONED_LOCK_ERR);
-                    let epoch_info = epoch_manager.get_epoch_info(&epoch_id)?;
+                    let epoch_info = epoch_manager
+                        .get_epoch_info(&epoch_id)
+                        .map_err(errors::WrappedQueryError::from)?;
                     (epoch_info.epoch_height, epoch_info.protocol_version)
                 };
 
-                match self.call_function(
-                    shard_id,
-                    *state_root,
-                    block_height,
-                    block_timestamp,
-                    prev_block_hash,
-                    block_hash,
-                    epoch_height,
-                    epoch_id,
-                    account_id,
-                    method_name,
-                    args.as_ref(),
-                    &mut logs,
-                    &self.epoch_manager,
-                    current_protocol_version,
-                    #[cfg(feature = "protocol_feature_evm")]
-                    self.evm_chain_id(),
-                ) {
-                    Ok(result) => Ok(QueryResponse {
-                        kind: QueryResponseKind::CallResult(CallResult { result, logs }),
-                        block_height,
-                        block_hash: *block_hash,
-                    }),
-                    Err(err) => Ok(QueryResponse {
-                        kind: QueryResponseKind::Error(QueryError { error: err.to_string(), logs }),
+                let call_function_result = self
+                    .call_function(
+                        shard_id,
+                        *state_root,
                         block_height,
-                        block_hash: *block_hash,
+                        block_timestamp,
+                        prev_block_hash,
+                        block_hash,
+                        epoch_height,
+                        epoch_id,
+                        account_id,
+                        method_name,
+                        args.as_ref(),
+                        &mut logs,
+                        &self.epoch_manager,
+                        current_protocol_version,
+                        #[cfg(feature = "protocol_feature_evm")]
+                        self.evm_chain_id(),
+                    )
+                    .map_err(errors::WrappedQueryError::from)?;
+                Ok(QueryResponse {
+                    kind: QueryResponseKind::CallResult(CallResult {
+                        result: call_function_result,
+                        logs,
```

Comment:
I am not quite sure it is the way we want to wrap things. Can we change the `CallResult(CallResult { result, log })` into 

```rust
struct QueryResponseKind {
   // ...
   CallResult(CallResult)
}

#[serde(tag = "execution_status", rename="SCREAMING_SNAKE_CASE")]
enum CallResult {
    Success { result, logs },
    Failure { error, logs },
}
```

I see it that the JSON RPC response for a failing view function call will be:

```json
{
  "jsonrpc": "2.0",
  "result": {
    "execution_status": "FAILURE",
    "error": ...,
    "logs": ...
  }
}
```

It does not make sense to me to have a failing CallResult as part of the QueryError since it is not the query failed, but the contract code.